### PR TITLE
Enhance Startup class conventions content

### LINKED ISSUE #7522: How to call StartupDevelopment class?
How to remove WebHostBuilder.UseStartup&lt;TStartup&gt; method in BuildWebHost(string[] args) in Program.cs in order to call StartupDevelopment class. The builder.UseStartup&lt;Startup&gt; method overrides this setting. However, I would like the program to automatically detect whether it should fire StartupDevelopment, StartupStaging or StartupProduction

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ceffa6a2-770c-1cd9-5145-dcd423f9dd22
* Version Independent ID: 1cb5eb30-dd84-e5e1-5826-d74243fabf04
* Content: [Use multiple environments in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-2.1)
* Content Source: [aspnetcore/fundamentals/environments.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/environments.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #7522 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/environments?view=aspnetcore-1.0&branch=pr-en-us-7525#environment-based-startup-class-and-methods)

### CODE REVIEW

File: aspnetcore/fundamentals/environments.md [hunk 1]
```diff
@@ -242,13 +242,121 @@ See [Configuration by environment](xref:fundamentals/configuration/index#configu
...
 
-When an ASP.NET Core app starts, the [Startup class](xref:fundamentals/startup) bootstraps the app. If a `Startup{EnvironmentName}` class exists, the class is called for that `EnvironmentName`:
+### Startup class conventions
 
-[!code-csharp[](environments/sample/EnvironmentsSample/StartupDev.cs?name=snippet&highlight=1)]
+When an ASP.NET Core app starts, the [Startup class](xref:fundamentals/startup) bootstraps the app. The app can define separate `Startup` classes for different environments (for example, `StartupDevelopment`), and the appropriate `Startup` class is selected at runtime. The class whose name suffix matches the current environment is prioritized. If a matching `Startup<EnvironmentName>` class isn't found, the `Startup` class is used.
 
-[WebHostBuilder.UseStartup&lt;TStartup&gt;](/dotnet/api/microsoft.aspnetcore.hosting.webhostbuilderextensions.usestartup#Microsoft_AspNetCore_Hosting_WebHostBuilderExtensions_UseStartup__1_Microsoft_AspNetCore_Hosting_IWebHostBuilder_) overrides configuration sections.
+To implement environment-based `Startup` classes, create a `Startup<EnvironmentName>` class for each environment in use and a fallback `Startup` class:
```
Comment (Tratcher) [inline aspnetcore/fundamentals/environments.md]: The use of angle brackets here is confusing, it looks like generics. Is there another seperator you can use? `|` (pipe)? `Startup|EnvironmentName|`

Comment (guardrex) [reply aspnetcore/fundamentals/environments.md]: I see what you mean. We use braces from time-to-time, and Rick did braces for this topic back in the day.

`Startup{EnvironmentName}`

Comment (Tratcher) [reply aspnetcore/fundamentals/environments.md]: Good enough.

File: aspnetcore/fundamentals/environments.md [hunk 2]
```diff
@@ -242,13 +242,121 @@ See [Configuration by environment](xref:fundamentals/configuration/index#configu
 
 ## Environment-based Startup class and methods
 
-When an ASP.NET Core app starts, the [Startup class](xref:fundamentals/startup) bootstraps the app. If a `Startup{EnvironmentName}` class exists, the class is called for that `EnvironmentName`:
+### Startup class conventions
 
-[!code-csharp[](environments/sample/EnvironmentsSample/StartupDev.cs?name=snippet&highlight=1)]
+When an ASP.NET Core app starts, the [Startup class](xref:fundamentals/startup) bootstraps the app. The app can define separate `Startup` classes for different environments (for example, `StartupDevelopment`), and the appropriate `Startup` class is selected at runtime. The class whose name suffix matches the current environment is prioritized. If a matching `Startup{EnvironmentName}` class isn't found, the `Startup` class is used.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/environments.md]: bootstrap is slang and doesn't MT. Make it a link or better yet don't use it. I think you're fine saying it load the app.

**Pull request merged**